public/app/features/dashboard/components/TransformationsEditor/TransformationOperationRow.tsx

```diff
@@ -60,6 +62,8 @@ export const TransformationOperationRow = ({
   // output of previous transformation
   const [prevOutput, setPrevOutput] = useState<DataFrame[]>([]);
 
+  const dynamicRefId = getTransformationDynamicRefId(uiConfig.id, data.series);
```

Comment:
This kind of sucks, to generate it here and then generate the same thing again in the individual transformation, but I'm not sure there's a great way to go from here to the transformer function without adding something to the object that comes from the saved dashboard JSON. I'll investigate adding it to the context instead.

Reply:
Okay, I tried to fix it by adding it to the context, but it breaks when you add the transformation list. Even changing it to show the right ref ID doesnt fix the filter. I can fix this if someone is really against this method, but if it's fine, it will be easier.